pkg/annotations/annotations.go

```diff
@@ -231,3 +232,12 @@ func ExtractResponseBuffering(anns map[string]string) (string, bool) {
 	s, ok := anns[AnnotationPrefix+ResponseBuffering]
 	return s, ok
 }
+
+// ExtractHostAliases extracts the host-aliases annotation value.
+func ExtractHostAliases(anns map[string]string) ([]string, bool) {
+	val, exists := anns[AnnotationPrefix+HostAliasesKey]
+	if val == "" {
+		return nil, exists
+	}
+	return strings.Split(val, ","), exists
+}
```

Comment:
In the meantime if the end-user does manage to set the annotation to an empty string for some odd reason though, we can just return false here as it will be semantically equivalent to what we're doing now, but would be a functional improvement:

```suggestion
// ExtractHostAliases extracts the host-aliases annotation value.
func ExtractHostAliases(anns map[string]string) ([]string, bool) {
	val, exists := anns[AnnotationPrefix+HostAliasesKey]
	if !exists {
		return nil, false
	}
	if val == "" {
		return nil, false
	}
	return strings.Split(val, ","), true
}
```